### ISSUE TITLE
debug ice if we pass superfluous args after `panic` to fn: ` no errors reported for args`

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#112437]
### Code

```Rust
#![feature(never_type)]
fn f(a: !) {}

fn main() {
    f(panic!(), 1);
}
```

### Meta

`rustc --version --verbose`:
```
<version>
```

### Error output

```
<output>
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
too long
```

</p>
</details>

**Comment by matthiaskrgr** [comment]
[ice.log.tar.gz](https://github.com/rust-lang/rust/files/11696786/ice.log.tar.gz)
